migrate_to_single_container.rb

```diff
@@ -0,0 +1,180 @@
+                :environment, :dry_run, :settings, :logger
+
+  def initialize(username)
+    @username = username
+
+    @environment = ENV["ENVIRONMENT"] || "staging"
+    @settings = YAML.load(File.read('config.yml'))[@environment]
+
+    @swift_host = @settings["swift"]["host"]
+    @swift_token = File.read("tmp/swift_token.txt").strip
+
+    @dry_run = ENV["DRYRUN"] || false # disables writing anything when true
+
+    @logger = Logger.new("log/migrate_to_single_container.log")
+    log_level = ENV["LOGLEVEL"] || "INFO"
+    logger.level = Kernel.const_get "Logger::#{log_level}"
+    logger.progname = username
+  end
+
+  def migrate
+    logger.info "Starting migration for '#{username}'"
+    set_container_migration_state("in_progress")
+    begin
+      work_on_dir("", "")
+    rescue Exception => ex
+      logger.error "Error migrating documents for '#{username}': #{ex}"
+      set_container_migration_state("not_started")
+      # write username to file for later reference
+      File.open('log/failed_migration.log', 'a') { |f| f.puts username }
+      exit 1
+    end
+    delete_container_migration_state
+    File.open('log/finished_migration.log', 'a') { |f| f.puts username }
+    logger.info "Finished migration for '#{username}'"
+  end
+
+  def is_dir?(name)
+    name[-1] == "/"
+  end
+
+  def set_container_migration_state(type)
+    redis.set("rs:container_migration:#{username}", type) unless dry_run
+  end
+
+  def delete_container_migration_state
+    redis.del("rs:container_migration:#{username}") unless dry_run
+  end
+
+  def work_on_dir(directory, parent_directory)
+    logger.debug "Retrieving listing for '#{parent_directory}#{directory}'"
+
+    listing = get_directory_listing_from_swift("#{parent_directory}#{directory}")
+
+    if listing
+      listing.split("\n").each do |item|
+        if is_dir? item
+          # get dir listing and repeat
+          work_on_dir(item, "#{parent_directory}")
+        else
+          copy_document("#{parent_directory}", item)
+        end
+      end
+    end
+  end
+
+  def copy_document(directory, document)
+    old_document_url = "#{url_for_directory(@username, directory)}/#{escape(document)}"
+    new_document_url = "#{new_url_for_directory(@username, directory)}/#{escape(document)}"
+
+    logger.debug "Copying document from #{old_document_url} to #{new_document_url}"
+
+    response = do_get_request(old_document_url)
+
+    unless dry_run
+      do_put_request(new_document_url, response.body, response.headers[:content_type])
+    end
+  end
+
+  def redis
+    @redis ||= Redis.new(@settings["redis"].symbolize_keys)
+  end
+
+  def get_directory_listing_from_swift(directory)
+    is_root_listing = directory.empty?
+
+    get_response = nil
+
+    do_head_request("#{url_for_directory(@username, directory)}") do |response|
+      return "" if response.code == 404
+
+      if is_root_listing
+        get_response = do_get_request("#{container_url_for(@username)}/?path=")
```

Comment:
@galfert Using `?path` does not work without having "folders" created as files in Orbit. So the migration worked for your user because you still had those folders but it doesn't work for my newly created development user. `?path=` returns an empty list if you don't have intermediate folders as files. This should use the `get_directory_listing_from_redis_via_lua` method like in the current liquor-cabinet